Azure/azure-cli
# [AppService] functionapp: Added support for in preview dotnet-isolated runtime.

**Description**


This is mainly a stacks update to add support for the new and in preview dotnet-isolated runtime. This is different from dotnet because it runs in a separate process from the functions host. I also had to add a bit of temporary custom code because of a known bug that's being fixed for creating dotnet-isolated linux consumption apps (linuxFxVersion needs to be blank in this scenario). It will be removed as soon as the bug is fixed.

**Testing Guide**


I added 3 news tests to make sure the CLI is able to create dotnet-isolated apps for a variety of scenarios (windows consumption, linux dedicated and consumption).

```
test_functionapp_on_linux_app_service_dotnet_isolated
test_functionapp_consumption_linux_dotnet_isolated
test_functionapp_windows_runtime_dotnet_isolated
```

**History Notes**


[AppService] az functionapp create: Added preview support for creating dotnet-isolated apps.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
